src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { FC, useContext, SyntheticEvent } from "react"
+import styled, { css, ThemeContext } from "styled-components"
+import { IStyled } from "common/types"
+import { oAccountCircle as UserProfileIcon } from "../Icon/Icons"
+
+interface AvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  size?: string
+  onClick?: (event?: SyntheticEvent) => void
+  picture?: string
+  name?: string
+  disabled: boolean
+}
+
+interface AvatarBaseProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  isPicture: boolean
+  isName: boolean
+  isIcon: boolean
+  size: string
+  disabled: boolean
+}
+
+const getThemeColor = (props, color) => {
+  const {
+    theme: {
+      knitui: { chromaPalette },
+    },
+  } = props
+  return chromaPalette[color].hex()
+}
+
+const getBoxShadowStyle = props => {
+  const {
+    theme: {
+      knitui: { shades },
+    },
+    customProps: { disabled },
+  } = props
+
+  return disabled ? "none" : `0px 0px 2px ${shades.blue50.hex()}`
+}
+
+const getFontSize = props => {
+  const {
+    customProps: { size },
+  } = props
+  if (size.includes("px")) {
+    const avatarSize = Number(size.split("px")[0])
+    return (0.14 * avatarSize) / 2.4
+  } else {
+    return 1.4
+  }
+}
+
+const getLineHeight = props => {
+  const {
+    customProps: { size },
+  } = props
+  if (size.includes("px")) {
+    const avatarSize = Number(size.split("px")[0])
+    return (0.2 * avatarSize) / 2.4
+  } else {
+    return 2
+  }
+}
+
+const getBackgroundColor = (props, state) => {
+  const { isPicture, isName, isIcon, disabled } = props.customProps
+  if (isPicture || isIcon) {
+    return "transparent"
+  } else if (isName) {
+    if (disabled || state === "default") {
```

Comment:
This `if` conditions seems to bre redundant, since the it will be covered in the last `else` block anyways.